### PR TITLE
Chore: blog post updates 2

### PR DESCRIPTION
**Pull request by fsansalvadore**
## What kind of change does this PR introduce?

Some more blog post updates.
https://zone-www-dot-com-git-fs-blog-post-updates-2-supabase.vercel.app/blog/designing-with-ai-midjourney#prompts-with-midjourney

## What is the new behavior?

- highlight Table of content (toc) active section while scrolling
- update `Quote` component
- add `Img` custom component with additional props:
  - **wide** - boolean / optional
  - **caption** - string / optional
  - **captionAlign** - "left" | "center" (default) | "right" / optional

## Before

Quotes:
Screenshot 2023-04-13 at 11 40 04

Screenshot 2023-04-13 at 11 39 53

Active toc section wasn't being highlighted.
And the image was always full width and without a caption.

## Additional context

Quote:
Screenshot 2023-04-13 at 11 36 09
Screenshot 2023-04-13 at 11 36 52

Toc active section (works with emojis too!):
Screenshot 2023-04-13 at 11 44 20

Image:

```tsx
<Img
  src="/images/blog/launch-week-7/designing-with-ai-midjourney/midjourney-variations.png"
  caption="Final AI generated images" // optional
  captionAlign="center" // optional and defaults to "center"
  wide={true} // optional
/>
```

Screenshot 2023-04-13 at 11 17 10

Screenshot 2023-04-13 at 11 20 17
Screenshot 2023-04-13 at 11 20 26

### CODE REVIEW

File: apps/www/styles/index.css [hunk 1]
```diff
@@ -85,14 +85,20 @@ h6:not(.overwrite) {
...
-  transform: translate3d(3px, 0, 0) !important;
-  color: hsla(153, 60%, 53%, 1) !important;
+  @apply !translate-x-1 !text-brand-900;
 }
 
 .prose-toc a {
   transition: all 0.07s ease-in-out;
 }
 
+.quote-figure p:first-child {
```
Comment (MildTomato) [inline apps/www/styles/index.css:95]: this can be put in tailwind config

Comment (MildTomato) [reply apps/www/styles/index.css:95]: in the typography setting. then the same styling can be used across all the apps.

**Review by MildTomato — CHANGES_REQUESTED** [review]

File: apps/www/styles/index.css (hunk 1 above)
Comment (MildTomato) [reply apps/www/styles/index.css:95]: https://github.com/supabase/supabase/blob/0a020247728b8a82a03115f4fba01bfede871bcd/packages/config/tailwind.config.js#L29-L164

**Comment by MildTomato** [comment]
image

Not sure how it will look, but you could try adding more padding above and below the quotes so they stand our a little more.
might look weird though. just a suggestion.

File: apps/www/styles/index.css (hunk 1 above)
Comment (fsansalvadore) [reply apps/www/styles/index.css:95]: This doesn't seem to work though... can't use class selectors here? @MildTomato 

Screenshot 2023-04-13 at 15 47 56

Comment (MildTomato) [reply apps/www/styles/index.css:95]: you did restart local ? you have there's no hot reloading of tailwind config

Comment (fsansalvadore) [reply apps/www/styles/index.css]: I thought I did, but I didn't! 😁

File: apps/www/components/Quote.tsx [hunk 2]
```diff
@@ -11,12 +11,17 @@ export default function Quote(props: Props) {
   const { children, caption, img } = props
 
   return (
-    <figure className="border border-gray-100 p-8 dark:border-gray-600">
-      <div className="flex flex-col gap-5">
-        <blockquote className="text-scale-1200 mb-4 border-none" style={{ margin: 0, padding: 0 }}>
+    <figure className="quote-figure p-0 mt-8">
```
Comment (MildTomato) [inline apps/www/components/Quote.tsx:14]: ahh i should of maybe said before...
you can write some (weird) tailwind for this that only me and you will understand 😅:

`![&_p:first-child]:mt-0 ![&_p:last-child]:mt-0 `

also (sorry if you already know this..) if your just trying to remove the margins added by 'prose', you can wrap the whole component in `.not-prose` to stop tailwind typography applying css in child elements.

**Review by MildTomato — APPROVED** [review]

**Pull request merged**